pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -193,6 +196,70 @@ type BMCDetails struct {
 	DisableCertificateVerification bool `json:"disableCertificateVerification,omitempty"`
 }
 
+// HardwareRAIDVolume defines the desired configuration of volume in hardware RAID
+type HardwareRAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.
+	// If unspecified or set be 0, the maximum capacity of disk will be used for logical disk.
+	// +kubebuilder:validation:Minimum=0
+	SizeGibibytes *int `json:"sizeGibibytes,omitempty"`
+
+	// RAID level for the logical disk. The following levels are supported: 0;1;1+0.
+	// +kubebuilder:validation:Enum="0";"1";"1+0"
+	Level string `json:"level" required:"true"`
+
+	// Name of the volume. Should be unique within the Node. If not specified, volume name will be auto-generated.
+	// +kubebuilder:validation:MaxLength=64
+	Name string `json:"name,omitempty"`
+
+	// Select disks with only rotational or solid-state storage
+	Rotational *bool `json:"rotational,omitempty"`
+
+	// Integer, number of disks to use for the logical disk. Defaults to minimum number of disks required
+	// for the particular RAID level.
+	// +kubebuilder:validation:Minimum=1
+	NumberOfPhysicalDisks *int `json:"numberOfPhysicalDisks,omitempty"`
+}
+
+// SoftwareRAIDVolume defines the desired configuration of volume in software RAID
+type SoftwareRAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.
+	// If unspecified or set be 0, the maximum capacity of disk will be used for logical disk.
+	// +kubebuilder:validation:Minimum=0
+	SizeGibibytes *int `json:"sizeGibibytes,omitempty"`
+
+	// RAID level for the logical disk. The following levels are supported: 0;1;1+0.
+	// +kubebuilder:validation:Enum="0";"1";"1+0"
+	Level string `json:"level" required:"true"`
+
+	// Name of the volume. Should be unique within the Node. If not specified, volume name will be auto-generated.
+	// +kubebuilder:validation:MaxLength=64
+	Name string `json:"name,omitempty"`
+
+	// A list of device hints, the number of item must be greater than or equal to 2.
+	// +kubebuilder:validation:MinItems=2
+	PhysicalDisks []RootDeviceHints `json:"physicalDisks,omitempty"`
+}
+
+// RAIDConfig contains the configuration that are required to config RAID in Bare Metal server
+type RAIDConfig struct {
+	// If true will remove existing raid configuration and set new raid configuration.
+	DeleteExisting bool `json:"deleteExisting,omitempty"`
```

Comment:
I don't remember why we want this to be optional behavior. Shouldn't we configure the host to match exactly the settings given in the API?

Reply:
Because [this](https://github.com/metal3-io/baremetal-operator/pull/292#discussion_r416182403) comment. If we needn't `DeleteExisting`, i can delete it.

Reply:
We already clean the volumes on deprovision, so what @kplimack requested in that comment is already not possible.
This flag is just not setting up RAID when it's set. Its effect is identical to not specifying a RAID config, so it's completely redundant.

Reply:
@zaneb Thanks for your reply, I have deleted `DeleteExisting` flag.